packages/components/src/internal/components/ontology/OntologyTreePanel.tsx

```diff
@@ -0,0 +1,113 @@
+import { TreeNode, TreePanel } from './TreePanel';
+import { ConceptModel } from './models';
+
+export class OntologyBrowserModel {
+    title: string;
+}
+
+export interface OntologyTreeProps {
+    model: OntologyBrowserModel;
+}
+
+class OntologyTreeNode implements TreeNode {
+    toggled: boolean = false;
+    expanded: boolean = false;
+    loading: boolean = false;
+    active: boolean = false;
+    name: string;
+    children: [TreeNode];
+    data?: ConceptModel;
+
+    constructor(values?: {[key:string]: any}) {
+        this.active = false;
+        this.loading = false;
+        this.expanded = false;
+        this.children = undefined;
+        Object.assign(this, values);
+    }
+}
+
+const testData = new OntologyTreeNode({
+    name: 'Breakfast',
+    children: [
+        new OntologyTreeNode({ name: 'crepes' }),
+        new OntologyTreeNode({
+            name: 'pancakes',
+            children: [
+                new OntologyTreeNode({ name: 'swedish' }),
+                new OntologyTreeNode({ name: 'Buttermilk' })
+            ],
+        }),
+        new OntologyTreeNode({ name: 'waffles' }),
+    ],
+    expanded: true,
+    toggled: true,
+    active: false,
+    loading: false,
+});
+
+// function useTreeData(data, subscribe) {
+//     const [treeData, setTreeData] = useState(data);
+//
+//     useEffect(() => {
+//         function handleDataChange(node, expansion, callback): void {
+//             if (node) {
+//                 node.toggled = expansion;
+//                 node.expanded = expansion;
+//                 setTreeData({ ...node });
+//             }
+//         }
+//
+//         subscribe(handleDataChange);
+//
+//         return () => {
+//             subscribe(null);
+//         };
+//     });
+//
+//     return treeData;
+// }
+
+export const OntologyTreePanel: FC<OntologyTreeProps> = props => {
+    // const { model, treeData, onNodeToggle } = props;
+    const { model, } = props;
+    const [toggleHandler, setToggleHandler] = useState();
+    const [treeData, setTreeData] = useState(testData);
+    // const treeData = useTreeData(testData, setToggleHandler);
+
+    // const onToggle = useCallback((node:any, expanded: boolean, callback?: () => any):void => {
+    //         node.toggled = expanded;
+    //         node.expanded = expanded;
+    //         setTreeData(node);
+    //     },
+    //     [setTreeData]
+    // );
+    //
+    const nodeClickHandler = useCallback(
+        (node:any, expanded:boolean, callback?:() => any): void => {
+            node.toggled = expanded;
+            setTreeData({ ...treeData });
+
+            if (callback) {
+                callback();
+            }
+        },
+        [treeData, setTreeData]
+    );
+
+    // useEffect(() => {
+    // }, [treeData, setTreeData]);
+
+    return (
+        <div className="ontology-browser-tree-container padding-right">
+            <TreePanel data={treeData} onToggle={nodeClickHandler} />
```

Comment:
I was actually hoping/wondering if we could just re-use (and likely make some slight edits to) the existing `FileTree` component in ui-components instead of creating a new usage of `<Treebeard/>` via this `<TreePanel/>` component. I know it has "file" in the name and several variable/function names within it, but it doesn't have a whole lot to do with files. FileTree is mostly just a wrapper over Treebeard to handle the data loading and node expand/selection. For example, if I comment out lines `153-157` in `FileTree.tsx` and then look at the storybook stories for it, it looks pretty much like the component we are looking for.